ruby-challenges.rb

```diff
@@ -11,25 +11,101 @@
 # Expected output: ['tea', 'water', 'soda water']
 
 
+# Pseudo Code 
+# create a method : contains_letter 
+# takes in two arguments (array, single_letter)
+# according to google, we can use method of .select on our array to iterate through each 'word'
+# we also need to evaluate whether each word includes the single_letter, and we can do this by invoking the method of .include? and pass in single_letter
+
+
+
+def contains_letter(array, single_letter)
+    array.select { |word| word.include? single_letter }
+  end
+
+# p contains_letter(beverages_array, letter_o)
+# p contains_letter(beverages_array, letter_t)
+
 # -------------------2) Create a method that takes in an array of numbers and returns the sum of the numbers. Use the test variables provided.
 
 nums_array1 = [42, 7, 27]
-# Expected output: 76
+# # Expected output: 76
 
 nums_array2 = [25, 17, 47, 11]
 # Expected output: 100
 
+# Pseudo Code 
+# Create a method sum_of 
+# input is an array 
+# our output must be the sum of the numbers in our array 
+# i recall in JS there was a built-in method of .reduce that would add up all the values in an array and output the total of those values---similarly, .inject(:+) in Ruby serves the same purpose. 
+
+def sum_of (nums_array)
+    nums_array.inject(:+)
+end
+
+# p sum_of nums_array1 
+# p sum_of nums_array2
+
+
 
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
 
+class Bike
```

Comment:
Solid work! Since you accomplished the goal, let's look some other ways to enhance the quality of your code. Line 62, since the question has given you the default value of 0 mph. You can hard code current speed value.
Line 59, since you have default values, you can just pass  one argument---->the model. Which will then change the structure of Line 69.
Since you are not introducing any new variables, you will not need to pass any arguments on your my_bike method. Nevertheless, your class method worked. I am simply sharing suggestions. Great work!